packages/assets-controllers/src/TokensController.ts

```diff
@@ -741,7 +741,13 @@ export class TokensController extends BaseController<
     const chainIdToAddTokens = detectionChainId ?? chainId;
 
     let newAllTokens = allTokens;
-    if (newTokens?.length) {
+    if (
+      newTokens?.length ||
+      (newTokens !== undefined &&
```

Comment:
Don't know about y'all but is this more obvious? Doesn't seem warranted to use different levels of checking for `newTokens` vs `allTokens` on the line below?
```suggestion
      (newTokens &&
```